packages/react-components/src/components/abstract-tree-output-component.tsx

```diff
@@ -4,14 +4,22 @@ import { ResponseStatus } from 'tsp-typescript-client/lib/models/response/respon
 
 export abstract class AbstractTreeOutputComponent<P extends AbstractOutputProps, S extends AbstractOutputState> extends AbstractOutputComponent<P, S> {
     renderMainArea(): React.ReactNode {
-        const treeWidth = this.props.widthWPBugWorkaround - this.getHandleWidth() - this.props.style.chartWidth;
+        // Make tree thiner when chart has a y-axis
+        const yAxisBuffer = this.props.outputDescriptor.type === 'TREE_TIME_XY' ? this.props.style.yAxisWidth: 0;
+        const treeWidth = this.props.widthWPBugWorkaround - this.getHandleWidth() - this.props.style.chartWidth - yAxisBuffer;
```

Comment:
Wouldn't it be possible to make it so that only xy-output-component has a y-axis? (and any other component that has a y-axis).

Because now the other components need to implement renderYAxis() when it is not relevant to them. In the future, what possible other part(s) of other components will they need to 'not' have?

Perhaps this main area should be split in renderLeft/renderRight instead of renderTree/renderChart, where the default left is a tree, the default right is a chart, and xy-chart can override renderLeft to add a y-axis to its left part. Then all others don't need to know about it.

Reply:
That still implicates changing the tree to be different, depending on whether there is a y-axis or not.
I tried a few alternatives, but it was hard to keep the vertical alignment of all charts.
I found the actual implementation was the easiest way to achieve the desired effect. As it satisfies the requirements of the ticket, I suggest it to be accepted.
We could open another where we can plan to experiment further options.